// svelte-media-queries/utils/MQLEvent.js
/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/ban-ts-comment */
import { Types } from "../components/MediaQuery.types";
import { getType } from "./getType";
export var MQLEventMethods;
(function (MQLEventMethods) {
    MQLEventMethods["add"] = "addEventListener";
    MQLEventMethods["remove"] = "removeEventListener";
})(MQLEventMethods || (MQLEventMethods = {}));
export class MQLEvent {
    static inline(item, handler = () => { }, method = MQLEventMethods.add) {
        (item && item[method]) && item[method]('change', handler);
    }
    static array(arr, handler = () => { }, method = MQLEventMethods.add) {
        //@ts-ignore
        arr.flat(Infinity).forEach(item => autoMQLEvent(item, handler, method));
    }
    static object(obj, handler = () => { }, method = MQLEventMethods.add) {
        for (const key in obj) {
            autoMQLEvent(obj[key], handler, method);
        }
    }
}
export function autoMQLEvent(mql, handler = () => { }, method = MQLEventMethods.add) {
    const type = getType(mql);
    if (mql instanceof MediaQueryList)
        return MQLEvent.inline(mql, handler, method);
    //@ts-ignore
    if (type === Types.array)
        return MQLEvent.array(mql, handler, method);
    //@ts-ignore
    if (type === Types.object)
        return MQLEvent.object(mql, handler, method);
}
